### PR TITLE
fix(markdown-it): convert missing namespace

### PR DESCRIPTION
**Pull request by Mister-Hope**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by Mister-Hope** [comment]
cc @runarberg

File: types/markdown-it/lib/parser_block.d.mts [hunk 1]
```diff
@@ -3,15 +3,13 @@ import Ruler from "./ruler.mjs";
 import StateBlock from "./rules_block/state_block.mjs";
 import Token from "./token.mjs";
 
-declare namespace ParserBlock {
-    type RuleBlock = (state: StateBlock, startLine: number, endLine: number, silent: boolean) => boolean;
-}
+export type RuleBlock = (state: StateBlock, startLine: number, endLine: number, silent: boolean) => boolean;
```
Comment (peterblazejewicz) [inline types/markdown-it/lib/parser_block.d.mts:6]: can you use something like:
```ts
declare type RuleBlockFn = (state: StateBlock, startLine: number, endLine: number, silent: boolean) => boolean;
export type RuleBlock = RuleBlockFn;

declare namespace ParserBlock {
    /**
     * @deprecated
     * Use:
     * ```js
     * import { RuleBlock } from "markdown-it/lib/parser_block.mjs"`
     * ```
     */
    type RuleBlock = RuleBlockFn;
}
```
that will not break existing code (still at 14.0). Not sure what is direct use-case requiring this rewrite

Comment (runarberg) [reply types/markdown-it/lib/parser_block.d.mts:6]: @peterblazejewicz  It was my mistake, I believe. I migrated the typedefs for this lib over to ES modules last week. As I did that I moved away from namespaces to directly exported symbols. I did it with all the other modules, but must have simply forgot this one.

Comment (Mister-Hope) [reply types/markdown-it/lib/parser_block.d.mts:6]: > @peterblazejewicz  It was my mistake, I believe. I migrated the typedefs for this lib over to ES modules last week. As I did that I moved away from namespaces to directly exported symbols. I did it with all the other modules, but must have simply forgot this one.

yes, this the point. only this is forgotten to be migrate.

namespace should be completely being removed in v14

Comment (runarberg) [reply types/markdown-it/lib/parser_block.d.mts:6]: Yes, I would agree this is a bug. My code dependant project only uses inline rules, so this particular module was never tested on my own code when I made the migration:

If I modify my code a bit the error is glaring:

```js
/**
 * @typedef {import("markdown-it/lib/parser_inline.mjs").RuleInline} RuleInline
 * @typedef {import("markdown-it/lib/parser_block.mjs").RuleBlock} RuleBlock
 * @typedef {import("markdown-it/lib/rules_inline/state_inline.mjs").default} StateInline
 */
```

`RuleInline` works fine, however `RuleBlock` fails to typecheck:

```
lib/ruler.js:3:57 - error TS2694: Namespace '"/my-lib/node_modules/@types/markdown-it/lib/parser_block"' has no exported member 'RuleBlock'.

3  * @typedef {import("markdown-it/lib/parser_block.mjs").RuleBlock} RuleBlock
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by Mister-Hope** [comment]
Ready to merge

**Pull request merged**